src/content/app/tools/blast/blast-download/submissionDownload.ts

```diff
@@ -0,0 +1,185 @@
+    │  ├─ sequence 2/
+    │  │  ├─ table.csv
+    │  │  ├─ alignments.txt
+
+ */
+
+type EnrichedBlastJobWithResults = BlastJobWithResults & {
+  csv: string;
+  raw: string;
+};
+
+type EnrichedBlastSubmission = BlastSubmission & {
+  results: EnrichedBlastJobWithResults[];
+};
+
+const downloadBlastSubmission = async (
+  submission: BlastSubmission,
+  dispatch: AppDispatch
+) => {
+  const blastedAgainst = submission.submittedData.parameters.database; // 'cdna' | 'dna' | 'pep'
+  const jobIds = submission.results.map(({ jobId }) => jobId);
+  const blastJobsQuery = dispatch(fetchAllBlastJobs.initiate(jobIds));
+  const rawBlastResultsQuery = dispatch(
+    fetchAllBlastRawResults.initiate(jobIds)
+  );
+  const { data: fetchedJobResults } = await blastJobsQuery;
+  const { data: fetchedRawJobResults } = await rawBlastResultsQuery;
+
+  blastJobsQuery.unsubscribe();
+  rawBlastResultsQuery.unsubscribe();
+
+  if (!fetchedJobResults || !fetchedRawJobResults) {
+    throw 'Failed to download all BLAST jobs';
+  }
+
+  const allBlastJobs = submission.results.map((job, index) => ({
+    ...job,
+    data: fetchedJobResults[index].result
+  }));
+
+  const allBlastJobsWithCSVs = allBlastJobs.map((job) => {
+    let csv = '';
+    if (blastedAgainst === 'dna') {
+      csv = createCSVForGenomicBlast(job.data);
+    } else if (blastedAgainst === 'cdna') {
+      csv = createCSVForTranscriptBlast(job.data);
+    } else if (blastedAgainst === 'protein') {
+      csv = createCSVForProteinBlast(job.data);
+    }
+
+    return {
+      ...job,
+      csv
+    };
+  });
+
+  const allBlastJobsWithRawData = fetchedRawJobResults.map((job, index) => {
+    const jobWithoutRawData = allBlastJobsWithCSVs[index];
+    return {
+      ...jobWithoutRawData,
+      raw: job.result
+    };
+  });
+
+  const zip = await createZipArchive({
+    ...submission,
+    results: allBlastJobsWithRawData
+  });
+
+  const blob = await zip.generateAsync({ type: 'blob' });
+  const zipFileName = getNameForZipRoot(submission);
+  await downloadBlobAsFile(blob, `${zipFileName}.zip`);
+};
+
+const createZipArchive = async (submission: EnrichedBlastSubmission) => {
+  const JSZip = await import('jszip').then((module) => module.default); // use a dynamic import to split this library off in a separate chunk
+
+  const {
+    submittedData: { species: allSpecies },
+    results
+  } = submission;
+  const speciesFolderNamesMap = new Map<string, string>();
+
+  for (const species of allSpecies) {
+    const folderName = `${species.scientific_name}-${species.genome_id}`;
+    speciesFolderNamesMap.set(species.genome_id, folderName);
+  }
+
+  const zip = new JSZip();
+  const rootFolder = zip.folder(getNameForZipRoot(submission));
+
+  for (const blastResult of results) {
+    const { genomeId, sequenceId, csv, raw } = blastResult;
+    const csvFileName = 'table'; // NOTE: this will probably change
+    const rawFileName = 'output'; // NOTE: this will probably change
```

Comment:
Should this be `alignments` as per L49?

Reply:
You mean the name of the file? Probably; but it contains more than just alignments... It's the raw output of the tool. We can call it alignments for now since this is what Lucy wanted, but it's more than just that.

Reply:
`output` is okay here.  It's just the comments above mention that this file will be called `alignments.txt`.

Reply:
The comment above is a very rough approximation of the contents of the files.

But I can rename the file to `alignments` here if it helps.

Reply:
No, don't worry. I just brought this up just to check if the file name is correct.